### PR TITLE
Emulate account entry extensions in simulation.

### PR DESCRIPTION
**Pull request by dmkozh**
### What

Emulate account entry extensions in simulation.

This also adds a general mechanism for modifying any ledger entries retrieved from the snapshot.

### Why

Improve simulation accuracy.

Fixes https://github.com/stellar/rs-soroban-env/issues/1357

### Known limitations

N/A

### CODE REVIEW

File: soroban-simulation/src/snapshot_source.rs [hunk 1]
```diff
@@ -115,3 +120,169 @@ impl<T: SnapshotSourceWithArchive> SnapshotSource for AutoRestoringSnapshotSourc
...
+    }
+}
+
+fn fill_account_ext_v2(account_ext_v1: &mut AccountEntryExtensionV1) {
+    match &mut account_ext_v1.ext {
+        AccountEntryExtensionV1Ext::V0 => {
+            let mut ext = AccountEntryExtensionV2 {
+                num_sponsored: 0,
+                num_sponsoring: 0,
+                signer_sponsoring_i_ds: Default::default(),
```
Comment (sisuresh) [inline soroban-simulation/src/snapshot_source.rs:267]:
```suggestion
                signer_sponsoring_ids: Default::default(),
```

Comment (sisuresh) [inline soroban-simulation/src/snapshot_source.rs:267]: In core, we resize this vector to the size of the signers vector on the account when the v2 extension is created - https://github.com/stellar/stellar-core/blob/f7a767ac5f6a2f64d9584a1042c57881a57586cf/src/transactions/TransactionUtils.cpp#L83. We should do the same here right?

Comment (dmkozh) [reply soroban-simulation/src/snapshot_source.rs:267]: That's not a typo, that's what we generate from the XDR (the field is named IDs).

Comment (sisuresh) [reply soroban-simulation/src/snapshot_source.rs:267]: Ahh yeah obviously :facepalm:

Comment (leighmcculloch) [reply soroban-simulation/src/snapshot_source.rs:267]: That's unfortunate way the name is being generated, and is a bug. We should be able to fix that in xdrgen.
- https://github.com/stellar/xdrgen/issues/194

Comment (dmkozh) [reply soroban-simulation/src/snapshot_source.rs]: I'm not 100% sure, but it's not hard to support this just in case. I've added the respective logic.

**Review by sisuresh — APPROVED** [review]

**Pull request merged**